capabilityquerier/pkg/capabilityquerier/capabilityquerier.go

```diff
@@ -72,32 +73,44 @@ func GetAndSendCapabilityStatement(ctx context.Context, args *map[string]interfa
 
 	var err error
 	message := Message{
-		URL: qa.FhirURL.String(),
+		URL: qa.FhirURLString,
 	}
 	metadataURL := endpointmanager.NormalizeEndpointURL(qa.FhirURL.String())
 	// Query fhir endpoint
 	err = requestCapabilityStatementAndSmartOnFhir(ctx, metadataURL, metadata, qa.Client, &message)
 	if err != nil {
-		log.Warnf("Got error:\n%s\n\nfrom URL: %s", err.Error(), qa.FhirURL.String())
-		message.Err = err.Error()
+		select {
+		case <-ctx.Done():
+			log.Warnf("Got error: server could not be reached from URL: %s", qa.FhirURLString)
+			message.Err = "server could not be reached from URL: " + metadataURL
+		default:
+			log.Warnf("Got error:\n%s\n\nfrom URL: %s", err.Error(), qa.FhirURLString)
+			message.Err = err.Error()
+		}
 	}
 
 	wellKnownURL := endpointmanager.NormalizeWellKnownURL(qa.FhirURL.String())
 	// Query well known endpoint
 	err = requestCapabilityStatementAndSmartOnFhir(ctx, wellKnownURL, wellknown, qa.Client, &message)
 	if err != nil {
-		message.Err = err.Error()
+		select {
+		case <-ctx.Done():
+			message.Err = "server could not be reached from URL: " + wellKnownURL
```

Comment:
This overwrites the error message from querying the metadata. I think we should either include both errors or only the error message from the /metadata endpoint. If the /metadata endpoint fails I believe the /.well-known/smart-configuration endpoint will also fail but I'm not 100% sure.  @mayerm94 Thoughts?

Reply:
I saw this as well in the original code and had the same thought as you, as I noticed that the well known endpoint error overwrites the metadata error or it puts an error in the db for the well known endpoint even if the metadata endpoint is fine. Would it make more sense to only look at the metadata endpoint for errors?

Reply:
Now that I think about it, I believe I added it myself. I think it would make sense to just add errors for the metadata.

Reply:
I removed added the error for the wellknown url, but lint still needed error checking for the request capability function for the well known endpoint, so I changed it so it just prints out a warning message like it does for the metadata url but it does not add this error to the message struct.